WSL: `code-insiders` 1.107.0-insider fails to start (`MODULE_NOT_FOUND` for `cli.js`)

<!-- ⚠️⚠️ Do Not Delete This! bug_report_template ⚠️⚠️ -->
<!-- Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/ -->
<!-- 🕮 Read our guide about submitting issues: https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions -->
<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Insiders build to see if your issue has already been fixed: https://code.visualstudio.com/insiders/ -->
<!-- 💡 Instead of creating your report here, use 'Report Issue' from the 'Help' menu in VS Code to pre-fill useful information. -->
<!-- 🔧 Launch with `code --disable-extensions` to check. -->
Does this issue occur when all extensions are disabled?: Yes/No

<!-- 🪓 If you answered No above, use 'Help: Start Extension Bisect' from Command Palette to try to identify the cause. -->
<!-- 📣 Issues caused by an extension need to be reported directly to the extension publisher. The 'Help > Report Issue' dialog can assist with this. -->
**Title:** WSL: `code-insiders` fails with `MODULE_NOT_FOUND` for `resources/app/out/cli.js` in 1.107.0-insider

**VS Code Version:** 1.107.0-insider
**Commit:** 2d69e225bfc64f111fdb9b9294a76bfbff1e99c0
**OS Version:** Windows 11 Enterprise 24H2 (Build 26100.7171) + WSL Ubuntu 22.04.5 LTS (jammy)
**Node.js (from error):** v22.21.1
**Edition:** VS Code Insiders for Windows, launched from WSL (`code-insiders`)

---

### Steps to Reproduce

1. Install VS Code Insiders 1.107.0-insider on Windows.

2. Open a WSL Ubuntu terminal.

3. Run:

   ```sh
   code-insiders .
   ```

4. Observe the error.

---

### Actual Behavior

`code-insiders` fails to start and prints:

```text
node:internal/modules/cjs/loader:1393
  throw err;
  ^

Error: Cannot find module 'C:\Users\<user>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\cli.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1390:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1032:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1037:22)
    at Module._load (node:internal/modules/cjs/loader:1199:37)
    at c._load (node:electron/js2c/node_init:2:17993)
    at TracingChannel.traceSync (node:diagnostics_channel:328:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:244:24)
    at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:186:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.21.1
```

---

### Expected Behavior

`code-insiders .` from WSL should launch VS Code Insiders on Windows and open the current folder.

---

### Additional Info

* In WSL, `which code-insiders` returns:

  ```text
  /mnt/c/Users/<user>/AppData/Local/Programs/Microsoft VS Code Insiders/bin/code-insiders
  ```

* The Windows install directory looks like this:

  ```text
  Microsoft VS Code Insiders\
    Code - Insiders.exe
    bin\
    2d69e225bf\
      resources\app\out\cli.js
    (no resources\ directory at the install root)
  ```

* So the WSL launcher appears to look for:

  ```text
  <install-root>\resources\app\out\cli.js
  ```

  but in this Insiders build the `resources` directory is under a versioned subfolder (`2d69e225bf\resources\app\out\cli.js`, where `2d69e225bf` matches the first 10 chars of the commit hash).

* This reproduces for me on two different machines and two different Windows versions and Ubuntu WSL distros.

--- a/resources/win32/insider/bin/code.sh
+++ b/resources/win32/insider/bin/code.sh
@@ -39,7 +39,7 @@ fi
 if [ $IN_WSL = true ]; then
 
 	export WSLENV="ELECTRON_RUN_AS_NODE/w:$WSLENV"
-	CLI=$(wslpath -m "$VSCODE_PATH/resources/app/out/cli.js")
+	CLI=$(wslpath -m "$VSCODE_PATH/$VERSIONFOLDER/resources/app/out/cli.js")
 
 	# use the Remote WSL extension if installed
 	WSL_EXT_ID="ms-vscode-remote.remote-wsl"

--- a/src/vs/server/node/server.cli.ts
+++ b/src/vs/server/node/server.cli.ts
@@ -279,7 +279,12 @@ export async function main(desc: ProductDescription, args: string[]): Promise<vo
 		} else {
 			const cliCwd = dirname(cliCommand);
 			const env = { ...process.env, ELECTRON_RUN_AS_NODE: '1' };
-			newCommandline.unshift('resources/app/out/cli.js');
+			if (process.platform === 'win32' && cliCommand.endsWith('Code - Insiders.exe')) {
+				const versionFolder = desc.commit.substring(0, 10);
+				newCommandline.unshift(`${versionFolder}/resources/app/out/cli.js`);
+			} else {
+				newCommandline.unshift('resources/app/out/cli.js');
+			}
 			if (verbose) {
 				console.log(`Invoking: cd "${cliCwd}" && ELECTRON_RUN_AS_NODE=1 "${cliCommand}" "${newCommandline.join('" "')}"`);
 			}

--- a/src/vs/server/node/server.cli.ts
+++ b/src/vs/server/node/server.cli.ts
@@ -279,8 +279,8 @@ export async function main(desc: ProductDescription, args: string[]): Promise<vo
 		} else {
 			const cliCwd = dirname(cliCommand);
 			const env = { ...process.env, ELECTRON_RUN_AS_NODE: '1' };
-			if (process.platform === 'win32' && cliCommand.endsWith('Code - Insiders.exe')) {
-				const versionFolder = desc.commit.substring(0, 10);
+			const versionFolder = desc.commit.substring(0, 10);
+			if (fs.existsSync(join(cliCwd, versionFolder))) {
 				newCommandline.unshift(`${versionFolder}/resources/app/out/cli.js`);
 			} else {
 				newCommandline.unshift('resources/app/out/cli.js');

